src/lib.rs

```diff
@@ -113,13 +115,25 @@ pub use miniscript::Miniscript;
 pub trait MiniscriptKey:
     Clone + Eq + Ord + str::FromStr + fmt::Debug + fmt::Display + hash::Hash
 {
+    /// Check if the publicKey is uncompressed. The default
+    /// implementation returns false for all Generic types of
+    /// MiniscriptKey, and returns the true only for
+    /// bitcoin::Publickey if the underlying key is uncompressed.
+    fn is_uncompressed(&self) -> bool {
+        false
+    }
+    /// The associated Hash type with the publicKey
     type Hash: Clone + Eq + Ord + str::FromStr + fmt::Display + fmt::Debug + hash::Hash;
 
     ///Converts an object to PublicHash
     fn to_pubkeyhash(&self) -> Self::Hash;
 }
 
 impl MiniscriptKey for bitcoin::PublicKey {
+    fn is_uncompressed(&self) -> bool {
+        !self.compressed
+    }
```

Comment:
...and we can add a doccomment here explaining that we are returning the compressedness flag of the bitcoin pubkey